src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs

```diff
@@ -309,6 +307,9 @@ internal static void WriteToStream(Stream input, Stream output, PSCmdlet cmdlet,
             catch (OperationCanceledException)
             {
             }
+            catch (AggregateException)
+            {
+            }
```

Comment:
We shouldn't hide download errors. Please remove.
```suggestion
```

Reply:
not hiding. 

plz see comment. https://github.com/PowerShell/PowerShell/pull/17896#issuecomment-1214674697

this change is to maintain existing behavior. without this change/ previously, if `copyTask` fails, it does not throw an exception (the method exits gracefully). 

With the new change, if `copyTask` fails during `copyTask.Wait()`, it will throw a new type of `AggregateException`.

Reply:
So looking at https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.wait?view=net-6.0 it seems that an AggregateException is thrown where the inner exception is a TaskCanceledException instead of a OperationCanceledException being thrown.  So it seems to me the right change here is to remove the catch for an OperationCanceledException and in the catch of the AggregateException, we should only ignore when InnerException is TaskCanceledException

Reply:
@SteveL-MSFT OperationCanceledException comes from `Wait()` cancellation (user press Ctrl-C) - this we should ignore, `AggregateException` comes from `copyTask` itself (lost network, server breaks connection and so on)  - this we should report.

Reply:
There are two task related exceptions that could occur in the method WriteToStream.
a. copyTask due to stream download or copy to file error. 
b. cancellationToken expires

The previous behavior is this, 
- For #a, no exception is thrown inside WriteToStream. Because, there is no wait on copyTask, only IsCompleted is checked (which doesn’t throw). 
- For #b, OperationCanceledException is thrown and suppressed. 

The new behavior (with fix) is this, 
- For #a, AggregateException exception is thrown inside WriteToStream. Because, there is wait on copyTask. 
- For #b, OperationCanceledException is thrown and suppressed.

I have caught AggregateException in addition to OperationCanceledException, to maintain previous behavior of WriteToStream. 

Here’s my dummy code to demonstrate the issue, [AbinzSharedTools/Program.cs at e9dbe47054d2b1a22b7e1b304ae237181e212c4a · AAATechGuy/AbinzSharedTools (github.com)](https://github.com/AAATechGuy/AbinzSharedTools/blob/e9dbe47054d2b1a22b7e1b304ae237181e212c4a/scripts/20220814_testPowerShellBugFix/Program.cs), output explained in this comment https://github.com/PowerShell/PowerShell/pull/17896#issuecomment-1214674697

Reply:
We can't ignore an AggregateException.  But we can look at what exceptions the AggregateException contains and selectively ignore that (such as a wait timeout exception).

https://docs.microsoft.com/en-us/dotnet/api/system.aggregateexception?view=net-6.0#examples

Reply:
I've removed `AggregateException`. 
But, please note, earlier `copyTask` errors didn't bubble up within `WriteToStream`. Now it does. That's the known behavior change.

Reply:
updated per discussion with Steve. 
I don't have a list of exceptions that could be safely caught to maintain backward compatibility, so keeping AggregateException.